sql-common/client.c

```diff
@@ -1878,6 +1878,190 @@ mysql_get_ssl_cipher(MYSQL *mysql __attribute__((unused)))
 }
 
 
+#if defined(HAVE_OPENSSL) && !defined(EMBEDDED_LIBRARY)
+
+#include <openssl/x509v3.h>
+
+#if defined(HAVE_X509_CHECK_HOST) && defined(HAVE_X509_CHECK_IP)
```

Comment:
I was also thinking about doing this but for testing purposes (to be able to easily switch between 'emulated' and 'native' implementations on platforms with newer OpenSSL) it's better to have a single define in a source file rather than move it to cmake level. I would prefer to leave it as is.